src/Illuminate/Queue/Worker.php

```diff
@@ -495,10 +479,10 @@ protected function raiseAfterJobEvent($connectionName, $job)
      *
      * @param  string  $connectionName
      * @param  \Illuminate\Contracts\Queue\Job  $job
-     * @param  \Exception  $e
+     * @param  \Throwable  $e
      * @return void
      */
-    protected function raiseExceptionOccurredJobEvent($connectionName, $job, $e)
+    protected function raiseExceptionOccurredJobEvent($connectionName, $job, Throwable $e)
```

Comment:
Can we add the `Job` type-hint here?

Reply:
Also for all other methods in this class.

Reply:
Yeh, I did actually add Job everywhere, then I changed my mind, because I think some of the phpdoc is wrong, or at least, it's not always being called with actual jobs. ;)

Reply:
I think we should leave the correction of Job phpdoc and typehints to a separate PR.

Reply:
@GrahamCampbell yep sounds good 👍